pyccel/codegen/printing/cwrappercode.py

```diff
@@ -273,88 +258,146 @@ def get_cast_function_call(self, cast_type, arg):
+
+        return numpy_type_check, python_type_check, error
+
     # -------------------------------------------------------------------
     # Functions managing  the creation of wrapper body
     # -------------------------------------------------------------------
-    def _body_optional_variable(self, tmp_variable, variable, collect_var, check_type = False):
+
+    def _valued_variable_management(self, variable, collect_var, tmp_variable):
         """
-        Responsible for collecting value and managing error and create the body
-        of optional arguments in format
-                if (pyobject == Py_None){
-                    collect Null
-                }else if(Type Check == False){
-                    Print TypeError Wrong Type
-                    return Null
-                }else{
-                    assign pyobject value to tmp variable
-                    collect the adress of the tmp variable
-                }
+        Responsible for creating the body collecting the default value of an valuedVariable
+        and the check needed.
+        if the valuedVariable is optional create body to collect the new value
+
         Parameters:
         ----------
         tmp_variable : Variable
             The temporary variable  to hold result
-        Variable : Variable
+        variable     : Variable
             The optional variable
-        collect_var : variable
+        collect_var  : Variable
             the pyobject type variable  holder of value
-        check_type : Boolean
-            True if the type is needed
 
         Returns
         -------
-        body : list
-            A list of statements
+        section      :
+            IfSection
+        collect_body :
+            body that collect new optional variable value
         """
-        body = [IfSection(PyccelEq(VariableAddress(collect_var), VariableAddress(Py_None)),
-                [Assign(VariableAddress(variable), Nil())])]
-        if check_type : # Type check
-            check = PyccelNot(PyccelOr(NumpyType_Check(variable, collect_var)
-                    , PythonType_Check(variable, collect_var)))
-            error = PyErr_SetString('PyExc_TypeError', '"{} must be {}"'.format(variable, variable.dtype))
-            body += [IfSection(check, [error, Return([Nil()])])]
-        body += [IfSection(LiteralTrue(), [self._create_collecting_value_body(variable, collect_var, tmp_variable),
-                    Assign(VariableAddress(variable), VariableAddress(tmp_variable))])]
-        body = [If(*body)]
 
-        return body
+        valued_var_check  = PyccelEq(VariableAddress(collect_var), VariableAddress(Py_None))
+        collect_body      = []
+
+        if variable.is_optional:
+            collect_body  = [Assign(VariableAddress(variable), VariableAddress(tmp_variable))]
+            section       = IfSection(valued_var_check, [Assign(VariableAddress(variable), Nil())])
```

Comment:
See above. Shouldn't this be an `AliasAssign`?

Reply:
done in 333a1a6

Reply:
done 
333a1a6